tests/fixture/tmpnet/network.go

```diff
@@ -540,11 +519,22 @@ func (n *Network) Stop(ctx context.Context) error {
 	return nil
 }
 
-// Restarts all nodes in the network.
+// Restarts all non-ephemeral nodes in the network.
 func (n *Network) Restart(ctx context.Context) error {
 	n.log.Info("restarting network")
-	for _, node := range n.Nodes {
-		if err := n.RestartNode(ctx, node); err != nil {
+	if err := restartNodes(ctx, n.Nodes...); err != nil {
+		return err
+	}
+	return WaitForHealthyNodes(ctx, n.log, n.Nodes...)
+}
+
+// Waits for the provided nodes to become healthy.
+func WaitForHealthyNodes(ctx context.Context, log logging.Logger, nodes ...*Node) error {
```

Comment:
nit it would be simpler code to not use variadic arguments and just use a slice of nodes, especially since all callers always use `nodesSlice...)` in the call